src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -815,16 +815,10 @@ public FunctionExpression GetReferenceExpression(ResourceMetadata resource, Synt
                     new JTokenExpression("full"));
             }
 
-            var shouldIncludeApiVersion =
-                !context.Settings.EnableSymbolicNames &&
-                (resource.IsExistingResource ||
-                (resource is DeclaredResourceMetadata { Symbol.DeclaringResource: var declaringResource } && declaringResource.HasCondition()));
-
-            if (shouldIncludeApiVersion)
+            if (!context.Settings.EnableSymbolicNames)
```

Comment:
Is this check still needed because a `reference(<symbolic name>)` template expression can't include an API version?

Reply:
Yeah, exactly.